### ISSUE TITLE
Chests and Enemies respawn when using QoL NG+

### ISSUE THREAD

**Issue by TheRamosOnline** [NiceneNerd/UKMM#174]
Mod: **https://gamebanana.com/mods/508732**

When using this specific mod, all chests and enemies respawn when reloading an area. Be it through reloading a save, dying and continuing or exiting a shrine.
Without the mod, chests despawn properly.
I reported this to the developer of the QoL NG+ mod and they said it didn't happen on their end using BCML.

I used the Hylian Trousers chest for testing, I could loot it, save and then reload then loot it again.

**Comment by OhNoOhNOOO** [comment]
Can concur, have the same issues myself. Enemies respawn despite there not being a blood moon and chests that should only be collectible once can be collected over and over by reloading the map.

**Comment by GingerAvalanche** [comment]
This would be due to flags getting removed from the mod completely. Please try with the latest UKMM and see if it still happens.

**Comment by Synby** [comment]
Late but would like to confirm that this still happens, both in the current release and the more up to date fork.

**Comment by GingerAvalanche** [comment]
This is now a year and a half out of date, and I've not paid total attention to how QoL NG+ has functioned on more recent releases. Does this still happen? I've talked to some people who have had it in their mod list who were reporting other bugs, and who have not said anything about this, so I assume it's fixed, but given that I didn't know in the first place if it was an all-the-time thing or subject to specific configurations (either of the mod's options or its being combined with other mods), I can't say one way or the other.

**Comment by MarcyMayCry** [comment]
Yes this is still a problem. It's quite disheartening. I haven't touched botw in many years and I was excited to come back and install mods. I spent all day modding the game just to find out this is the case. If you reload, everything respawns. I even noticed a weird issue where mushrooms respawned visually but I wasn't able to pick them up, and that happened after the first time the game made me open the slate, not from a reload. Not sure if that was related, but I'd assume so.  (edit: clarification this is with latest release build 0.17.1) ((actually edit once again: this should be the latest commit I believe as I cloned the repo and ran cargo build --release. I think this is how it works anyways, Im still learning git.))

EDIT 3: I fixed it by disabling the "alternate shops" section of the mod. I found this out because of a comment from 2 years ago on gamebanana that states "i tested the individual mods that make up this, it looks like its caused by Alternate shops". The mod dev stated 2 years ago "The mod works fine with BCML, you need to report this issue on the UKMM github so they can address it." - I didn't see anyone here on github saying anything about the shops so I don't know if this was known to this day. 

~~Alternate shops has a patch file called "Quality of Life - Merchant Override" to work with BCML but Quality of Life - Merchant Override is a .zip file and cant be installed into UKMM to my knowledge? Either way it exists to make sure that "BCML correctly merges and sorts the changes made to NPC merchant inventories with other mods that do the same". Im not sure if somehow getting that into UKMM would even fix the issue though. (I dont think it would as the author states "I am told that UKMM does merge this file" - so I think this file is only needed for BCML?)~~

It's nice that I know the problem, but it sucks that I will be missing out on a feature. I would love to know if this is fixable. (If I am needed to do or show anything, I would gladly help test anything btw to help this get fixed)

**Comment by GingerAvalanche** [comment]
Re: Merchant override - it can be installed. UKMM can install any file types BCML can: bnp, zip, rar, 7z, extracted folder by dragging in a rules.txt or info.json, the works.

Installing the merchant override wouldn't solve this, though, that's just a set of changes to NPC actor packs, while the things governing respawn flags are all in the game data, about as far away as you can get, file-wise, while still being in the same game.

I'll look into Alternate Shops and see if I can figure out why it would do that. A few authors have had the "it works fine with BCML" attitude, which is unfortunate, because BCML was a pile of spaghetti that didn't work right (don't worry, I can say that, Nicene and I were the authors of BCML, too), so restricting everyone to using that is quite unfortunate.

EDIT: I've checked the alternate shops options against Second Wind (which does the same kind of thing wrt adding/deleting game data flags) and it appears to do everything exactly the same, which at the very least means it's not the mod's fault, directly. I'll look into why UKMM treats Second Wind and QoL NG+ differently at some point. Maybe tomorrow, if I have the energy.

**Comment by MarcyMayCry** [comment]
Thanks for the reply and info!

Yes I've seen plenty of mod authors say not to use UKMM but I think given time UKMM will prove itself. I'm probably going to make an AUR page for UKMM once the AUR opens back up the registering of accounts so that people on Linux can have an easier time installing it since I see so many comments from people having issues or not knowing how. Hopefully that gets more people to use it.

**Comment by GingerAvalanche** [comment]
I think if I can figure out why the settings validator passes invalid paths, I can fix most of the issues people have with UKMM. That’s been on my list forever and I’ve never gotten round to it.

It confuses me to hear that people can’t figure out how to install UKMM on Linux, it’s literally “extract executable, run executable.” About as easy as it’s possible to be. I’m almost hesitant to put it on the AUR since that would require to them to use the command line (GASP!), though the benefit of having a dev package that sits on the master branch head would be nice
